### PR TITLE
add gendoc --check, use in CI

### LINKED ISSUE #174: Run `gendoc` as a pass/fail as part of CI
Alá `rustfmt` or `clippy`, gives feedback without making edits.

### PR DESCRIPTION
**Pull request by conradoplg**
Closes #174 

This also fixes gendoc to work with secp256k1 (another reason why we need this PR :laughing: )

This is failing on purpose to show it works. After it's reviewed I'll commit the updated files

### CODE REVIEW

**Comment by dconnolly** [comment]
Failed?

File: frost-secp256k1/src/lib.rs [hunk 1]
```diff
@@ -268,21 +268,101 @@ pub mod keys {
...
+        use super::*;
+
+        /// The secret package that must be kept in memory by the participant
+        /// between the first and second parts of the DKG protocol (round 1).
+        ///
+        /// # Security
+        ///
+        /// This package MUST NOT be sent to other participants!
+        pub type Round1SecretPackage = frost::keys::dkg::Round1SecretPackage<S>;
+
```
Comment (dconnolly) [inline frost-secp256k1/src/lib.rs:298]: At this point, should this live in its own file (`dkg.rs`) ?

**Comment by conradoplg** [comment]
@dconnolly I ran gendoc and committed the result, and it's now passing as expected

File: frost-secp256k1/src/lib.rs (hunk 1 above)
Comment (dconnolly) [reply frost-secp256k1/src/lib.rs:298]: Tracked in: https://github.com/ZcashFoundation/frost/issues/185

**Review by dconnolly — APPROVED** [review]
🎉

**Pull request merged**